pycbc/distributions/constraints.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (C) 2017 Christopher M. Biwer
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""
+This modules provides classes for evaluating multi-dimensional constraints.
+"""
+
+from pycbc import transforms
+
+class Constraint(object):
+    """ Creates a constraint that evaluates to True if parameters obey
+    the constraint and False if they do not.
+    """
+    name = "custom"
+    required_parameters = []
+    def __init__(self, variable_args, **kwargs):
+
+        # set any given attributes and get transforms from variable_args
+        # to required parameters
+        for kwarg in kwargs.keys():
+            setattr(self, kwarg, kwargs[kwarg])
+        _, self.transforms = transforms.get_common_cbc_transforms(
+                                       self.required_parameters, variable_args)
```

Comment:
Is it necessary to store the common cbc transforms and apply them below? This is only ever used in inference runs. I think it's reasonable to expect that contraints could only be applied to either sampling parameters or variable args. If some other derived coordinate is desired, then a transforms section could be added by the user to the config file to ensure that parameter is in the params dictionary that is passed to the prior evaluator class. If you remove having to create the transforms then you can get rid of the variable args input parameter, and just require that the constraints be initalized with a function string to apply. Also, it would save on re-applying all of these transforms at each iteration in the MCMC (although that's probably not a very expensive operation).